services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -4721,17 +4726,16 @@ void preCheckStorePartitionCountUpdate(String clusterName, Store store, int newP
   }
 
   private void generateAndUpdateRealTimeTopicName(Store store) {
-    // get oldRealTimeTopicName from the store config because that will be more (or equally) recent than any version
-    // config
-    String oldRealTimeTopicName = Utils.getRealTimeTopicNameFromStoreConfig(store);
-    String newRealTimeTopicName = Utils.createNewRealTimeTopicName(oldRealTimeTopicName);
-    PubSubTopic newRealTimeTopic = getPubSubTopicRepository().getTopic(newRealTimeTopicName);
-
-    if (getTopicManager().containsTopic(newRealTimeTopic)) {
-      throw new VeniceException("Topic " + newRealTimeTopic + " should not exist.");
-    }
+    String newRealTimeTopicName = Utils.isRTVersioningApplicable(store.getName())
+        ? store.getName() + "_v" + (store.getLargestUsedRTVersionNumber() + 1) + Version.REAL_TIME_TOPIC_SUFFIX
+        : DEFAULT_REAL_TIME_TOPIC_NAME;
 
     store.getHybridStoreConfig().setRealTimeTopicName(newRealTimeTopicName);
+    // after partition count update, new VT, if is hybrid, has to use a new RT with the same partition count, so we
+    // increase
+    // `largestUsedRTVersionNumber`; this will ensure that a new RT is created when/if needed
+    store.setLargestUsedRTVersionNumber(store.getLargestUsedRTVersionNumber() + 1);
+    LOGGER.info("updated largestUsedRTVersionNumber to " + store.getLargestUsedRTVersionNumber());
```

Comment:
Here’s a cleaned-up and more professional version:

Two notes about logging:
	1.	Let's consistently use interpolation format for all log statements.
	2.	Always include sufficient information to provide context during troubleshooting. Without the RT name, store name, and cluster name, this log message in its current form is not useful.